src/app/ReadClient.h

```diff
@@ -131,14 +138,30 @@ class ReadClient : public Messaging::ExchangeDelegate
         virtual void OnSubscriptionEstablished(SubscriptionId aSubscriptionId) {}
 
         /**
-         * OnResubscriptionAttempt will be called when a re-subscription has been scheduled as a result of the termination of an
-         * in-progress or previously active subscription. This object MUST continue to exist after this call is completed. The
+         * OnResubscriptionNeeded will be called when a subscription that was started with SendAutoResubscribeRequest has terminated
+         * and re-subscription is needed. The termination cause is provided to help inform subsequent re-subscription logic.
+         *
+         * The base implementation automatically re-subscribes at appropriate intervals taking the termination cause into account
+         * (see ReadClient::DefaultResubscribePolicy for more details). If the default implementation doesn't suffice, the logic of
+         * ReadClient::DefaultResubscribePolicy is broken down into its constituent methods that are publicly available for
+         * applications to call and sequence.
+         *
+         * If the method is over-ridden, it's the application's responsibility to take the appropriate steps needed to eventually
+         * call-back into the ReadClient object to schedule a re-subscription (by invoking ReadClient::ScheduleResubscription).
+         *
+         * If the application DOES NOT want re-subscription to happen on a particular invocation of this method, returning anything
+         * other than CHIP_NO_ERROR will terminate the interaction and result in OnError, OnDeallocatePaths and OnDone being called
+         * in that sequence.
+         *
+         * This object MUST continue to exist after this call is completed. The
          * application shall wait until it receives an OnDone call to destroy the object.
          *
          * @param[in] aTerminationCause The cause of failure of the subscription that just terminated.
-         * @param[in] aNextResubscribeIntervalMsec How long we will wait before trying to auto-resubscribe.
```

Comment:
So in the new setup, how does a ReadClient::Callback that does not want to manage its own resubscribe timing get informed about what the selected timing is?  Is it by calling `ComputeTimeTillNextSubscription()`?  Might be worth documenting that.

Reply:
Isn't this text sufficient?

> If the default implementation doesn't suffice, the logic of ReadClient::DefaultResubscribePolicy is broken down into its constituent methods that are publicly available for applications to call and sequence.

If you go to that method, it has further documentation on the two methods it calls, and so on.